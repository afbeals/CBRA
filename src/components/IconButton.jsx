// Extenal
import React from 'react';
import PropTypes from 'prop-types';
import MaterialIconButton from '@material-ui/core/IconButton';

// Local

// Constants
const classname = 'iconButton';
/* Declarations */
/* Functions */

// Component
const IconButton = ({
  children,
  className,
  primary,
  ripple,
  size,
  ...rest
}) => {
  const props = {
    color: primary ? 'primary' : 'secondary',
    disableRipple: !ripple,
    size,
    ...rest,
  };
  return (
    <MaterialIconButton
      {...props}
      data-testid="iconButton"
      className={`${className ? `${className} ` : ''}${classname}`}
      role="button"
    >
      {children}
    </MaterialIconButton>
  );
};

IconButton.propTypes = {
  children: PropTypes.node.isRequired,
  primary: PropTypes.bool,
  className: PropTypes.string,
  ripple: PropTypes.bool,
  size: PropTypes.oneOf(['small', 'medium']),
};

IconButton.defaultProps = {
  primary: true,
  ripple: true,
  size: 'small',
};

export default IconButton;

export { IconButton };
