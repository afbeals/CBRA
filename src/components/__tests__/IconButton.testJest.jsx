// External
import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';

// Local
import IconButton from '../IconButton';

// tests
describe('IconButton', () => {
  afterEach(cleanup);
  it('should match IconButton to snapshot', () => {
    const { getByTestId } = render(<IconButton />);
    const iconButton = getByTestId('iconButton');
    expect(iconButton).toMatchSnapshot();
  });

  it('Should call the passed function', () => {
    const fnClick = jest.fn();
    const { getByTestId } = render(<IconButton onClick={fnClick} />);
    const iconButton = getByTestId('iconButton');
    fireEvent.click(iconButton);
    expect(fnClick).toHaveBeenCalled();
  });

  it('should render a disabled Button', () => {
    const { getByTestId } = render(<IconButton disabled />);
    const iconButton = getByTestId('iconButton');
    expect(iconButton.disabled).toEqual(true);
  });
});
