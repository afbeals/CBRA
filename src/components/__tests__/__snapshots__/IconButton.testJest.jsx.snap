// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match IconButton to snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiIconButton-root iconButton MuiIconButton-colorPrimary MuiIconButton-sizeSmall"
  data-testid="iconButton"
  role="button"
  tabindex="0"
  type="button"
>
  <span
    class="MuiIconButton-label"
  />
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
